Source/Dafny/DafnyAst.cs

```diff
@@ -2223,7 +2223,7 @@ public class BigOrdinalType : BasicType
     public override string TypeName(ModuleDefinition context, bool parseAble) {
       return "ORDINAL";
     }
-    public override bool Equals(Type that) {
+    public override bool Equals(Type that, bool keepConstraints = false) {
       return that.IsBigOrdinalType;
```

Comment:
To handle a type like `type MyOrdinal = x: ORDINAL | x != 8`, I think this needs to be:

    return that.NomalizeExpand(keepConstraints) is BigOrdinalType;

Reply:
OK